src/components/DateSelector/index.js

```diff
@@ -0,0 +1,80 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { makeStyles } from '@material-ui/core/styles';
+import DateFnsUtils from '@date-io/date-fns';
+import {
+  MuiPickersUtilsProvider,
+  KeyboardDatePicker,
+} from '@material-ui/pickers';
+import { styles } from './style';
+
+const useStyles = makeStyles(styles);
+
+export const DateSelector = ({
+  name,
+  label,
+  defaultValue,
+  onFocus,
+  helperText,
+  error,
+  disabled,
+}) => {
+  const [selectedDate, setSelectedDate] = useState(null);
+  const classes = useStyles();
+
+  useEffect(() => {
+    if (defaultValue) {
+      if (error) {
+        setSelectedDate(null);
+      } else {
+        setSelectedDate(defaultValue);
+      }
+    }
+  }, []);
+
+  return (
+    <MuiPickersUtilsProvider utils={DateFnsUtils}>
+      <KeyboardDatePicker
+        name={name}
+        value={selectedDate}
+        onChange={setSelectedDate}
+        onAccept={onFocus}
+        error={error}
+        disabled={disabled}
+        label={label}
+        variant="dialog"
+        format="MM/dd/yyyy"
+        helperText={helperText}
```

Comment:
hm, what happens when `helperText` is an array?

Reply:
good point! It will render every message in array. We'll pass only the first one to component